Services/AssessmentQuestion/classes/class.ilAsqFactory.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Class ilAsqQuestionAuthoringFactory
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+class ilAsqFactory
+{
+	/**
+	 * @param integer $parentObjectId
+	 * @return array
+	 */
+	public function getQuestionDataArray($parentObjectId)
```

Comment:
This is needed for the ilTableGUI, isn't it? I'm afraid of having this on the public interface since an array as far to unstructured and one could use this for whatever reason. how about passing the table to let id be "filled" with data internally?